application/helpers/admin/import_helper.php

```diff
@@ -2160,6 +2174,34 @@ function checkWrongQuestionAttributes($questionId)
     }
 }
 
+/**
+ * Upgrades question attributes 'random_order' and 'alphasort' into 'answer_order' as needed.
+ * @param int $questionId
+ */
+function upgradeAnswerOrderQuestionAttributes($questionId)
+{
+    $questionId = (int) $questionId;
+    $db = Yii::app()->db;
+    $db->createCommand(
+        "INSERT INTO {{question_attributes}} (qid, " . $db->quoteColumnName("attribute") . ", " . $db->quoteColumnName("value") . ")
+         SELECT qa.qid, 'answer_order' " . $db->quoteColumnName("attribute") . ", 'random' " . $db->quoteColumnName("value") . "
+         FROM {{question_attributes}} qa
+         JOIN {{questions}} q ON qa.qid = q.qid
+         WHERE " . $db->quoteColumnName("attribute") . " = 'random_order' AND " . $db->quoteColumnName("value") . " = '1' AND q.type IN ('!', 'L', 'O', 'R')
```

Comment:
Must test if random_order is not set in all DB

Reply:
Sorry, not following... :)

Reply:
In the lss imported : `random_order` can not exist. If it not exist : no issue in mariadb or mysql, surely pgsql but what with mssql :)

Reply:
And more : 2 request for each question more … see https://bugs.limesurvey.org/view.php?id=18479

Must update without DB reques
t
Something like https://github.com/LimeSurvey/LimeSurvey/blob/174a12431a6149272d3bb52772993d49fd610642/application/helpers/admin/import_helper.php#L435 


```
    if ($insertdata['attribute'] == 'random_order' && !empty($insertdata['value']) ) {
        $insertdata['attribute'] = 'answer_order';
        $insertdata['value'] = 'random';
    }
    if ($insertdata['attribute'] == 'alphasort' && !empty($insertdata['value']) ) {
        $insertdata['attribute'] = 'answer_order';
        $insertdata['value'] = 'alphabetical';
    }
```

Reply:
> In the lss imported : `random_order` can not exist. If it not exist : no issue in mariadb or mysql, surely pgsql but what with mssql :)

I don't see why this would fail. See this.
https://stackoverflow.com/a/45174111/1392917

Reply:
> Must update without DB request

The whole ticket is about combining 2 attributes into a new one.
Attribute questions are processed one by one and we don't know the order on which they are processed.
So, what you say would work for almost all cases. 

But if Alpha = ON and Random = ON (which I think is a possible comibination right now), the current behaviour is order=Random. We must keep the compatibility for that.

If alpha is processed after random, with the approach you sent, the order would be set to alpha, which doesn't match the current behaviour.

I would like to avoid querying, but that would involve a huge code reorganziation:
- Have atributes to be saved to be stored temp on an array 
- Review the Alpha + Random transformation on memory
- Save all at once

Not sure if it is worth that reorganization for 1 extra db query per question.

Reply:
> I would like to avoid querying, but that would involve a huge code reorganziation:

Maybe , but doing 2 request when upload : you surely broke import LSS.

> If alpha is processed after random, with the approach you sent, the order would be set to alpha, which doesn't match the current behaviour.

You can use a var and check it, quick sample

```
    $attribute_answer_order = null;
    foreach ($xml->question_attributes->rows->row as $row) {
    …
    …
        /* Fix answer_order */
        if ($insertdata['attribute'] == 'random_order' && !empty($insertdata['value']) ) {
            $insertdata['attribute'] = 'answer_order';
            $attribute_answer_order = $insertdata['value'] = 'random';
        }
        if ($attribute_answer_order != "random" && $insertdata['attribute'] == 'alphasort' && !empty($insertdata['value']) ) {
            $insertdata['attribute'] = 'answer_order';
            $attribute_answer_order = $insertdata['value'] = 'alphabetical';
        }
    …
    …
    }
```

Reply:
> Maybe , but doing 2 request when upload : you surely broke import LSS.

I believe you are being a bit too dramatic. But let me review workarounds.

> You can use a var and check it, quick sample

Is not that simple. If not, I would have already done it.

Reply:
> 2 extra queries per question will broke the import process?

If www and DB in in same server : no. But on different server … possible … see : https://bugs.limesurvey.org/view.php?id=18472

> Is not as simple as you wrote.

Oh yes need an array `$attribute_answer_order = [];` `$attribute_answer_order[$insertdata['qid']] = $insertdata['value'] = 'random';`
Maybe ?

Memory is easy to add, move DB is less easy.

Reply:
I will come up with something